### PR TITLE
Adds FrameCounter to smurf file emulator

### PR DESCRIPTION
**Pull request by jlashner**
Populates the FrameCounter array in the smurf_file_emulator data.

## Description

Populates the FrameCounter array in the smurf_file_emulator data

## Motivation and Context

load_smurf relies on this to linearize timestamps, and this is useful for the bookbinder.

## How Has This Been Tested?

Ran test functions locally

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Unless I am preparing a release, I have opened this PR onto the `develop` branch.

### CODE REVIEW

File: agents/smurf_file_emulator/smurf_file_emulator.py [hunk 1]
```diff
@@ -186,6 +194,8 @@ def get_data_frame(self, start, stop):
 
         times = np.arange(start, stop, 1. / self.sample_rate)
         nsamps = len(times)
+        frame_counter = np.arange(self.sample_num, self.sample_num + nsamps, dtype=int)
+        self.sample_num += nsamps
```
Comment (BrianJKoopman) [inline agents/smurf_file_emulator/smurf_file_emulator.py:198]: Is `self.sample_num` going to be used elsewhere later?

File: .gitignore [hunk 2]
```diff
@@ -136,3 +136,5 @@ key.*
 node.pid
 
 .DS_Store
+
+.vscode
```
Comment (BrianJKoopman) [inline .gitignore:140]: Move this to the vscode block starting at line 126.

**Review by BrianJKoopman — COMMENTED** [review]
Looks good, just one question below.

File: agents/smurf_file_emulator/smurf_file_emulator.py (hunk 1 above)
Comment (jlashner) [reply agents/smurf_file_emulator/smurf_file_emulator.py:198]: `get_data_frame` can be called multiple times per streaming session, so `self.sample_num` needs to be persistent / kept up-to-date

Comment (BrianJKoopman) [reply agents/smurf_file_emulator/smurf_file_emulator.py:198]: Ah, gotcha, thanks!

**Review by BrianJKoopman — APPROVED** [review]

**Pull request merged**